packages/astro/src/core/routing/manifest/generator.ts

```diff
@@ -2,6 +2,20 @@ import type { AstroConfig, RoutePart } from '../../../@types/astro.js';
 
 import { compile } from 'path-to-regexp';
 
+function sanitizeParams(params: Record<string, string | number | undefined>): Record<string, string | number | undefined> {
+	return Object.entries(params).reduce((acc, [key, value]) => {
```

Comment:
You could use `.map` instead of `.reduce`. It's easier to read, and you don't need to maintain an accumulator

Reply:
changed it to using map now. Considered simplifying like this, but final thought is that its not worth making an intermediate object:
```
function sanitizeParams(params: Record<string, string | number | undefined>): Record<string, string | number | undefined> {
	const sanitizedParams: Record<string, string | number | undefined> = {};

	Object.entries(params).forEach(([key, value]) => {
		if (typeof value === 'string') {
			sanitizedParams[key] = value
				.normalize()
				.replace(/#/g, '%23')
				.replace(/\?/g, '%3F');
		} else {
			sanitizedParams[key] = value;
		}

	})

	return sanitizedParams
}
```